include/mbedtls/md.h

```diff
@@ -107,6 +115,7 @@ typedef struct mbedtls_md_context_t {
     void *MBEDTLS_PRIVATE(hmac_ctx);
 } mbedtls_md_context_t;
 
```

Comment:
Unrelated to this line, but rather about this file:
Don't we want to guard the function declarations with `MBEDTLS_MD_LIGHT`?

Reply:
We usually don't guard function declarations in header files with the corresponding implementation macro. For example, before this PR they were not guarded with `MBEDTLS_MD_C`. Technically, the guards I'm adding wouldn't be necessary, but I think they're helpful to document what's part of MD-light and what's not.